### PR TITLE
Update getting advanced USB working on new swadge

### PR DESCRIPTION
**Pull request by cnlohr**
### Description
 * reboot-from-bootloader-into-swadge functionality in `tools/bootload_reboot_stub/`
 * reboot-to-bootloader functionality in `tools/reboot_into_bootloader/`
 * printf / ESP_LOG-over-usb in `tools/swadgeterm`
 * A sandbox demo in `tools/sandbox_test`
 * Add Windows support to USB stack for custom messages.

### CODE REVIEW

**Comment by AEFeinstein** [comment]
> Note: there are doxygen warnings for me anyway? Not sure what is going on there.

I built the docs in this branch and didn't get any warnings. What version of doxygen do you have?
```
PS D:\git\Swadge-IDF-5.0> doxygen --version
1.9.6
```

**Comment by AEFeinstein** [comment]
Also the emulator fails to build because this file isn't mocked (among maybe other issues)
```
main/swadge2024.c:141:10: fatal error: advanced_usb_control.h: No such file or directory
  141 | #include "advanced_usb_control.h"
      |          ^~~~~~~~~~~~~~~~~~~~~~~~
```

File: sdkconfig [hunk 1]
```diff
@@ -858,8 +858,7 @@ CONFIG_ESP_SYSTEM_ALLOW_RTC_FAST_MEM_AS_HEAP=y
 #
 # Memory protection
 #
-CONFIG_ESP_SYSTEM_MEMPROT_FEATURE=y
-CONFIG_ESP_SYSTEM_MEMPROT_FEATURE_LOCK=y
+# CONFIG_ESP_SYSTEM_MEMPROT_FEATURE is not set
```
[the remark this answers is not in the document]
Comment (cnlohr) [reply sdkconfig:861]: Yes.  It prevents most of the USB functionality I'm after, and, with basically no benefit.  It has no perf implications either.  It just prevents self-modifying code.

File: components/hdw-usb/hdw-usb.c [hunk 2]
```diff
@@ -151,15 +151,15 @@ uint16_t tud_hid_get_report_cb(uint8_t instance, uint8_t report_id,
...
     }
     else if (report_id == 172)
     {
-        return handle_advanced_usb_terminal_get(reqLen, buffer);
+        return handle_advanced_usb_terminal_get(reqLen + 1, buffer - 1);
     }
     else if (report_id == 173 && advancedUsbHandler)
     {
-        return advancedUsbHandler(buffer, reqLen, 1);
+        return advancedUsbHandler(buffer - 1, reqLen + 1, 1);
```
[the remark this answers is not in the document]
Comment (cnlohr) [reply components/hdw-usb/hdw-usb.c:162]: I can make them all buffer, len.  Or do you prefer the other way?

[the remark this answers is not in the document]
Comment (AEFeinstein) [reply components/hdw-usb/hdw-usb.c]: I don't have a strong preference one way or the other, I just like consistency.

**Review by AEFeinstein — APPROVED** [review]

**Comment by AEFeinstein** [comment]
As a note, this still needs instructions written up.

**Pull request merged**